### PR TITLE
#72 Added config for running ColumnMap-based jobs. Also added support…

### PR DESCRIPTION
**Pull request by rjrudin**
Overview:

1. New MigrateColumnMapsConfig for performing ColumnMap->XML migrations from a relational database
1. New OptionParserConfigurer interface. Implementing this allows a Config class to list its options when a user runs the too with "--help". 
1. Lots of files moved - I moved all the SQL-specific stuff out of core and into jobs. The ColumnMap stuff is still in core.

### CODE REVIEW

File: core/src/main/java/com/marklogic/spring/batch/Main.java [hunk 1]
```diff
@@ -43,14 +46,38 @@ public static void main(String[] args) throws Exception {
...
         OptionSet options = parser.parse(args);
-        ConfigurableApplicationContext ctx = buildApplicationContext(options);
-        JobParameters params = buildJobParameters(options);
-        JobLauncher launcher = getJobLauncher(ctx);
-        Job job = getJobToExecute(ctx, options);
-        try {
-            return launcher.run(job, params);
-        } finally {
-            ctx.close();
+        if (options.has(Options.HELP)) {
```
Comment (rjrudin) [inline core/src/main/java/com/marklogic/spring/batch/Main.java]: Added support for printing help

File: core/src/main/java/com/marklogic/spring/batch/Main.java [hunk 2]
```diff
@@ -62,18 +89,25 @@ public JobExecution runJob(String[] args) throws Exception {
...
+        parser.accepts(Options.CONFIG, "The fully qualified classname of the Spring Configuration class to register").withRequiredArg().required();
         parser.accepts(Options.JOB, "The name of the Spring Batch Job bean to run").withRequiredArg();
+        parser.accepts(Options.CHUNK_SIZE, "The Spring Batch chunk size").withRequiredArg();
 
         parser.accepts(Options.JOB_REPOSITORY_HOST, "Hostname of the MarkLogic Server for the JobRepository").withRequiredArg();
         parser.accepts(Options.JOB_REPOSITORY_PORT, "Port number of the App Server for the JobRepository. The App Server must not be SSL-enabled.").withRequiredArg().ofType(Integer.class).defaultsTo(8000);
         parser.accepts(Options.JOB_REPOSITORY_USERNAME, "The MarkLogic user to authenticate as against JobRepository App Server").withRequiredArg().defaultsTo("admin");
         parser.accepts(Options.JOB_REPOSITORY_PASSWORD, "The password for the JobRepository MarkLogic user").withRequiredArg();
 
+        parser.accepts(Options.JDBC_DRIVER, "Driver class name for connecting to a relational database").withRequiredArg();
```
Comment (rjrudin) [inline core/src/main/java/com/marklogic/spring/batch/Main.java]: Conferred with Scott - these JDBC options are now "standard" ones, as there can be a lot of reuse for them

File: core/src/main/java/com/marklogic/spring/batch/configuration/AbstractMarkLogicBatchConfig.java [hunk 3]
```diff
@@ -43,4 +50,10 @@ protected DatabaseClient getDatabaseClient() {
     public BatchConfigurer jobRepositoryConfigurer() {
         return new JobRepositoryConfigurer();
     }
+
+    public Integer getChunkSize() {
```
Comment (rjrudin) [inline core/src/main/java/com/marklogic/spring/batch/configuration/AbstractMarkLogicBatchConfig.java]: Defaulting chunk size to 10; can be overridden via command line too

File: jobs/build.gradle [hunk 4]
```diff
@@ -8,10 +8,16 @@ targetCompatibility = 1.8
 
 dependencies {
     compile fileTree(include: ['*.jar'], dir: 'lib')
-    compile "com.marklogic:marklogic-spring-batch:0.3.0"
 
     testCompile "com.marklogic:ml-junit:2.6.0"
     testCompile "org.springframework.batch:spring-batch-test:3.+"
+
+    // For testing SQL readers with an embedded database
```
Comment (rjrudin) [inline jobs/build.gradle]: Moved these dependencies here from core

**Comment by sastafford** [comment]
@rjrudin I'm fine with the changes.  This is going to require a manual merge due to some changes I made in Main.java that I just merged into dev.  Would you mind merging and closing out the pull request?

**Comment by rjrudin** [comment]
I merged this in by merging this branch locally into a new branch off dev, then merging that new branch in. Still have the issue where I think we need ml-gradle to setup a "fixed" app server / database for tests.

**Pull request closed without merging**